manually-adding-services-to-metrics.adoc

```diff
@@ -0,0 +1,127 @@
+
+=== Adding a new scraping configuration to the Prometheus instance
+
+In this example we will edit the existing Prometheus configuration to
+include a new `scrape_config` targetting our service.
+
+This example will target the Grafana instance deployed alongside
+Prometheus, which exposes a metrics endpoint in the default `/metrics`
+HTTP endpoint.
+
+See the
+link:https://prometheus.io/docs/prometheus/latest/configuration/configuration/\#scrape_config[Prometheus
+documentation on scrape_config] for more details.
+
+=== Via the OpenShift Console
+
+1.  Select your project in the `My Projects` sidebar
+
+image:img/manually-adding-services-to-metrics/myprojects.png[My
+projects]
+
+2.  Navigate to Prometheus' Config Map via
+`Resources > Config Maps > prometheus-configmap`
+
+image:img/manually-adding-services-to-metrics/configmap.png[Config Map]
+
+3.  Enter editing mode through `Actions > Edit YAML`
+
+image:img/manually-adding-services-to-metrics/edit.png[Edit YAML]
+
+4.  Include a new `scrape_config` targetting the Grafana service:
+
+[source,yaml]
+----
+apiVersion: v1
+data:
+  prometheus.yml: |-
+    ---
+    global:
+      evaluation_interval: 5s
+      scrape_interval: 5s
+    scrape_configs:
+      - job_name: prometheus
+        static_configs:
+          - targets:
+            - "localhost:9090"
+      - job_name: grafana
+        metrics_path: "/metrics" # default value
+        static_configs:
+          - targets:
+            - "grafana-internal:80"
+        # other configuration
+----
+
+The `target` of the config should point to the internally-resolvable
+name of the service, visible by navigating to
+`Applications > Services > grafana-internal` and has the format
+`{Hostname}:{Target Port}`.
+
+NOTE: The `grafana` service requires authentication through the OpenShift OAuth proxy, thus preventing connection from the unauthenticated Prometheus client, so the `grafana-internal` should be our target instead.
+
+5.  Trigger a manual deployment of the `prometheus` service to inject
+the updated Config Map by going to
+`Applications > Deployments > prometheus` and clicking `Deploy`.
+
+6. Verify the new target by navigating to the list of targets in Prometheus at https://prometheus-{project_name}.192.168.37.1.nip.io/targets, or by acessing the Prometheus web interface and navigating to `Status > Targets`:
+
+image:img/manually-adding-services-to-metrics/grafana-target.png[Grafana Target]
+
+=== Via the `oc` cli
+
+Run the following commands:
+
+[source,bash]
+----
+oc login # login to the OpenShift Cluster
+oc get projects # list available Projects
+oc use myproject # switch to the project where the prometheus and the custom service are located
+oc get configmaps # list available Config Maps
+oc edit configmap prometheus-configmap # open the prometheus configmap in $OC_EDITOR or $EDITOR
```

Comment:
As a side note I've had some troubles with `oc edit`-ing the prometheus configmap due to linebreaks coming up as `\n`